js/components/addressBook/addressBook_controller.js

```diff
@@ -4,8 +4,22 @@ angular.module('contactsApp')
 
 	ctrl.showUrl = false;
 	/* globals oc_config */
+
+	function compareVersion(version1, version2) {
+		var result=false;
+		for(var i=0;i<(Math.max(version1.length, version2.length));i++) {
```

Comment:
shouldn't it be Math.min? You don't want to access an index greater than the string length, do you? :-)